tests/test_downloadermiddleware_retry.py

```diff
@@ -104,5 +104,86 @@ def _test_retry_exception(self, req, exception):
         self.assertEqual(req, None)
 
 
+class MaxRetryTimesTest(unittest.TestCase):
+    def setUp(self):
+        self.crawler = get_crawler(Spider)
+        self.spider = self.crawler._create_spider('foo')
+        self.mw = RetryMiddleware.from_crawler(self.crawler)
+        self.mw.max_retry_times = 2
+
+    def test_without_metakey(self):
+        
+        req = Request('http://www.scrapytest.org/invalid_url')
+
+        # SETTINGS: RETRY_TIMES is NON-ZERO
+        self.mw.max_retry_times = 5
+        self._test_retry(req, DNSLookupError('foo'), 5)
+
+        # SETTINGS: RETRY_TIMES = 0
+        self.mw.max_retry_times = 0
+        self._test_retry(req, DNSLookupError('foo'), 0)
+
+        # RESET RETRY_TIMES SETTINGS
+        self.mw.max_retry_times = 2
```

Comment:
@redapple , thanks! I just read through the [TestCase.tearDown()](https://docs.python.org/2/library/unittest.html#unittest.TestCase.tearDown) docs.

In that case, `tearDown()` would just reset ` self.mw.max_retry_times = 2` - which is also set in `setUp()` as well. So don't you think `tearDown()` is not neccesary for this particular situation?